src/oop/special_class_method/hard/q7/MainTest.java

```diff
@@ -0,0 +1,338 @@
+                {value2, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] getMethodModifiers = {"public"};
+        String incorrectGetMethodMessage = String.join(" ",
+                "Your", getIntMethodName, "method does not correctly get the value of the", intFieldName, "attribute.");
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getIntMethodName, null), String.format(incorrectMethodDefinition, getDoubleMethodName, className));
+        assertTrue(testClass.hasModifier(getIntMethodName, null, "public"), String.format(incorrectModifierMessage, getDoubleMethodName, className));
+        assertTrue(testClass.hasReturnType(getIntMethodName, null, int.class), String.format(incorrectReturnType, getDoubleMethodName, className));
+        Object getMethodOutput = testClass.callMethod(getIntMethodName, getMethodModifiers, classInstance);
+        _assertEquals(value2, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    @Tag("dependent1")
+    public void piggyBankClassHasCorrectToStringMethod(double value, int b) throws Throwable {
+        Object[][] arguments = {
+                {value, double.class},
+                {b, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod("toString", null), String.format(incorrectMethodDefinition, "toString", className));
+        assertTrue(testClass.hasModifier("toString", null, "public"), String.format(incorrectModifierMessage, "toString", className));
+        assertTrue(testClass.hasReturnType("toString", null, String.class), String.format(incorrectReturnType, "toString", className));
+        String[] methodModifiers = {"public"};
+        String expected = "I have " + value + " dollars in me after " + b + " days.";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, "toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
```

Comment:
I think we should also have a test that checks if the methods work with the parameterless constructor.